detectors/aws/ecs/ecs.go

```diff
@@ -90,9 +103,82 @@ func (detector *resourceDetector) Detect(ctx context.Context) (*resource.Resourc
 		semconv.ContainerIDKey.String(containerID),
 	}
 
+	if len(metadataURIV4) > 0 {
+		containerMetadata, err := ecsmetadata.GetContainerV4(ctx, &http.Client{})
+		if err != nil {
+			return empty, err
+		}
+		attributes = append(
+			attributes,
+			semconv.AWSECSContainerARNKey.String(containerMetadata.ContainerARN),
+		)
+
+		taskMetadata, err := ecsmetadata.GetTaskV4(ctx, &http.Client{})
+		if err != nil {
+			return empty, err
+		}
+
+		clusterArn := taskMetadata.Cluster
+		if !strings.HasPrefix(clusterArn, "arn:") {
+			baseArn := containerMetadata.ContainerARN[:strings.LastIndex(containerMetadata.ContainerARN, ":")]
+			clusterArn = fmt.Sprintf("%s:cluster/%s", baseArn, clusterArn)
+		}
+
+		logAttributes, err := detector.getLogsAttributes(containerMetadata)
+		if err != nil {
+			return empty, err
+		}
+
+		if len(logAttributes) > 0 {
+			attributes = append(attributes, logAttributes...)
+		}
+
+		attributes = append(
+			attributes,
+			semconv.AWSECSClusterARNKey.String(clusterArn),
+			semconv.AWSECSLaunchtypeKey.String(strings.ToLower(taskMetadata.LaunchType)),
```

Comment:
This is [an enumeration](https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/resource/cloud_provider/aws/ecs.yaml#L18-L27) that does not allow custom values.  Can you use one of the [defined values](https://github.com/open-telemetry/opentelemetry-go/blob/main/semconv/v1.12.0/resource.go#L224-L229) for this key?

Reply:
Done in 8e6bf31d

Reply:
This has not been changed, only the tests.  This does not ensure that a non-standard value cannot be emitted.

Reply:
Ah, now I understood what you mean. Done in 49156c98.